dev/fattest.simplicity/src/componenttest/topology/impl/JavaInfo.java

```diff
@@ -126,7 +129,7 @@ private JavaInfo() {
         String vendor = System.getProperty("java.vendor").toLowerCase();
         if (vendor.contains("openj9"))
             VENDOR = Vendor.OPENJ9;
-        else if (vendor.contains("ibm") || vendor.contains("j9"))
+        else if (detectIBMJava())
             VENDOR = Vendor.IBM;
```

Comment:
So, are we changing the definition of what `Vendor.IBM` means to be something other than "The vendor is IBM"?

It feels like this is one place where the code to set `VENDOR` looks correct?

Reply:
The vendor String is not consistent enough and should not be used to infer available function. In another PR I want to get rid of this test class and try to use the version of JavaInfo from the kernel (where the vendor() method is now deprecated). Since vendor should only now be used for logging/debug, the `Vendor` enum isn't so useful any more ... might as well just return the String as provided by the JVM.

Reply:
So essentially your answer is that tests still rely on the result of this method to determine what function is available?

Whereas in the product code you've changed code to look for the existence of required classes than for particular JVM identifiers, in the test code you're happy to hack around it and have this method set `VENDOR` based on available function, rather than the actual vendor of the JVM?